lib/segment/analytics/request.rb

```diff
@@ -32,6 +32,11 @@ def initialize(options = {})
         http.open_timeout = 4
 
         @http = http
+
+        # If `start` is not called, Ruby adds a 'Connection: close' header
+        # too all requests, preventing us from reusing a connection for multiple
+        # HTTP requests
+        @http.start
```

Comment:
I added this comment because it took me a while to figure this out, and the Ruby docs don't mention it explicitly.

I'll also try raising this on the Ruby issue tracker once I've experimented a bit.